### ISSUE TITLE
AbstractDifferentiation

### ISSUE THREAD

**Issue by ParadaCarleton** [EnzymeAD/Enzyme.jl#349]
Does Enzyme have support for the [AbstractDifferentiation.jl](https://github.com/JuliaDiff/AbstractDifferentiation.jl/issues) interface? If not, is it planned?

**Comment by vchuravy** [comment]
It does not currently. But we would welcome a contribution to add it!

**Comment by vchuravy** [comment]
They will have to eventually invert their dependency structure... Adding all AD packages as a dependency is unecessary heavy for an interface package and the AD frameworks should implement their interface.

**Comment by ParadaCarleton** [comment]
> They will have to eventually invert their dependency structure... Adding all AD packages as a dependency is unecessary heavy for an interface package and the AD frameworks should implement their interface.

Thus why I opened the issue here; I believe the use of `@requires` to include AD package glue code in AbstractDiff is meant to be temporary. @mohamed82008 is this correct? In addition, what would be required to get Enzyme.jl working with the AbstractDiff interface?

**Comment by mohamed82008** [comment]
Yes this is the plan. Last I checked Enzyme, it didn't work with AbstractDifferentiation internals. I am working on a redesign of AbstractDifferentiation that should make it easier for Enzyme to support it.

**Comment by schlichtanders** [comment]
@mohamed82008 is there any update on AbstractDifferentiation supporting Enzyme or a respective roadmap/issue?

**Comment by mohamed82008** [comment]
Last I tried, Enzyme couldn't differentiate some of AbstractDifferentiation's nested closures. A redesign of AbstractDifferentiation is long due but I haven't had the time to complete it so far. Perhaps I can help someone else willing to take on that task by having a call to discuss some ideas.
